packages/http-client-csharp/generator/Microsoft.TypeSpec.Generator.ClientModel/src/Providers/MrwSerializationTypeDefinition.cs

```diff
@@ -359,7 +359,7 @@ internal MethodProvider BuildPersistableModelCreateCoreMethod()
                 ? MethodSignatureModifiers.Private
                 : MethodSignatureModifiers.Protected | MethodSignatureModifiers.Virtual;
 
-            if (_shouldOverrideMethods)
+            if (_shouldOverrideMethods && _model.BaseModelProvider is not SystemObjectTypeProvider)
```

Comment:
Should this logic go into `_shouldOverrideMethods`?

Reply:
it can't, details is in https://github.com/microsoft/typespec/pull/7008#discussion_r2059890761